kernel/kernel-api/src/main/java/io/delta/kernel/internal/SnapshotImpl.java

```diff
@@ -119,9 +120,10 @@ public StructType getSchema() {
 
   @Override
   public Optional<String> getDomainMetadata(String domain) {
-    return Optional.ofNullable(getDomainMetadataMap().get(domain))
-        .filter(dm -> !dm.isRemoved()) // only consider active domain metadatas (not tombstones)
-        .map(DomainMetadata::getConfiguration);
+    Optional<DomainMetadata> domainMetadata =
+        Optional.ofNullable(getDomainMetadataMap().get(domain));
+    domainMetadata.ifPresent(dm -> checkArgument(!dm.isRemoved()));
```

Comment:
do we really want to throw an exception? what are the semantics of this API?

could we, for exampl, just define them to return a non-removed domain metadata?

should callers first check if the domain is removed _before_ calling this API? how would they do that?

Reply:
This should return empty if it is removed -- that's the semantics outlined in the interface method docs + the current behavior

Reply:
But I think actually that should be an impossible state to reach? So maybe `checkArgument` is just the wrong use here -- `checkState` or something? or maybe we don't need to verify at all

Reply:
this api will only return active domain.

and yes this is an impossible state to reach after the the change. It will be reached only if internal impl has a bug. 
Will change it to checkState.